mayastor/src/bdev/nexus/mod.rs

```diff
@@ -59,3 +59,17 @@ pub fn nexus_instance_new(name: String, size: u64, children: Vec<String>) {
     let list = instances();
     list.push(Nexus::new(&name, size, None, Some(&children)));
 }
+
+/// called during shutdown so that the bdevs for each child is destroyed first
+pub async fn nexus_destroy_all() {
+    debug!("destroying all nexuses...");
+    for nexus in instances() {
```

Comment:
Could we use [`futures::join_all`](https://docs.rs/futures/0.3.12/futures/future/fn.join_all.html) here? So we could start them in parallel instead of waiting for each?

Reply:
I guess that's possible as a future enhancement but it appears that the existing SPDK behaviour (based on the log messages) is to destroy them sequentially so I'm mostly preserving existing behaviour here, just re-ordering which subsystems to destroy first.